### PR TITLE
Add/gm19 webauthn -- challenge & response pieces

### PR DESCRIPTION
**Pull request by jblz**
Targeting the branch in #36069

#### Changes proposed in this Pull Request

* Client-side pieces for calling the challenge and authenticate endpoints 

#### Testing instructions

* Register at least one hardware key using this branch (or the one in #36069)
* Run this branch
* Log out if you're not already
* Log in with your test user's user_id & password, (but don't submit a two-factor OTP code if one is set for the account)
* Click the button to use a hardware key
* Press the button on your hardware key
  * It should log you in

---

* Repeat the above with an unregistered key
  * It should not log you in

### CODE REVIEW

File: client/lib/webauthn/index.js [hunk 1]
```diff
@@ -183,7 +184,77 @@ function register() {
...
+		form.append( 'dev_hostname', window.location.hostname );
+	}
+	return fetch( `https://wordpress.com/wp-login.php?action=${ action }`, {
+		method: 'POST',
+		body: form,
+		credentials: 'include',
+	} ).then( response => response.json() );
+}
+
+function authenticate( wpcom_user_id, nonce ) {
```
Comment (jblz) [inline client/lib/webauthn/index.js]: @xknown:  It's not obvious to me where this is called (even though I definitely see this call going out.  Is it implicit somewhere or am I just too tired to find it :)

Comment (xknown) [reply client/lib/webauthn/index.js]: The `authenticate` method isn't not used anymore :), it was "deprecated" in 99a9581. We can probably remove it

Comment (xknown) [reply client/lib/webauthn/index.js]: Removed it in 8141ef4

**Comment by jblz** [comment]
Cool!  Thanks for all the great code, @Automattic/ex-owners 
I'm going to go ahead and merge this into the main feature branch and we can continue there.

**Pull request merged**